lib/bolt/plugin/terraform.rb

```diff
@@ -82,9 +82,18 @@ def load_statefile(opts)
 
       # Uses the Terraform CLI to pull remote state files
       def load_remote_statefile(opts)
-        stdout_str, stderr_str, = Open3.capture3('terraform state pull', chdir: opts['dir'])
+        dir = File.expand_path(opts['dir'])
+
+        begin
+          stdout_str, _stderr_str, status = Open3.capture3('terraform state pull', chdir: dir)
+        rescue StandardError => e
```

Comment:
I think we can rescue an `Errno::ENOENT` and assume it is due to no terraform executable on the path. The error message can be something to that effect. 

I am not sure what kinds of status's or errors that `terraform state pull` can give. I was just using the "local" backend:
```
terraform {
  backend "local" {
    path = "test/terraform_local.tfstate"
  }
}
```
Did you have an s3 backend set up? What happens if there is a connection or auth errror?

Reply:
I do. If there's a connection or auth error then terraform raises an error that we catch after the rescue block and also writes to `stderr`.

Reply:
Rescuing `Errno::ENOENT` works fine. The only issue with saying it's due to no terraform executable on the path is that the same kind of error is raised when the path to the `dir` does not exist. Would we want to validate the `dir` before attempting to load the state file in that case?